climpred/classes.py

```diff
@@ -385,36 +385,47 @@ def get_uninitialized(self):
         """Returns the xarray dataset for the uninitialized ensemble."""
         return self._datasets['uninitialized']
 
-    def smooth(self, smooth_kws='goddard2013'):
+    def smooth(self, smooth_kws=None, how='mean', **xesmf_kwargs):
         """Smooth all entries of PredictionEnsemble in the same manner to be
         able to still calculate prediction skill afterwards.
 
         Args:
-          xobj (xarray object):
-            decadal prediction ensemble output.
-
-        Attributes:
             smooth_kws (dict or str): Dictionary to specify the dims to
-                smooth compatible with `spatial_smoothing_xesmf`,
-                `temporal_smoothing` or `spatial_smoothing_xrcoarsen`.
+                smooth compatible with
+                :py:func:`~climpred.smoothing.spatial_smoothing_xesmf` or
+                :py:func:`~climpred.smoothing.temporal_smoothing`.
                 Shortcut for Goddard et al. 2013 recommendations:
-                'goddard2013'
-
-        Example:
-        >>> PredictionEnsemble.smooth(smooth_kws={'time': 2,
-            'lat': 5, 'lon': 4'})
-        >>> PredictionEnsemble.smooth(smooth_kws='goddard2013')
+                'goddard2013'. Defaults to None.
+            how (str): how to smooth temporally. From ['mean','sum']. Defaults to
+                'mean'.
+            **xesmf_kwargs (args): kwargs passed to
+                :py:func:`~climpred.smoothing.spatial_smoothing_xesmf`
+
+        Examples:
+            >>> PredictionEnsemble.smooth({'lead': 2, 'lat': 5, 'lon': 4'})
+            >>> PredictionEnsemble.smooth('goddard2013')
+            >>> PredictionEnsemble.smooth({'lon':1, 'lat':1}, method='patch')
+            >>> PredictionEnsemble.smooth({'lead':2}, how='sum')
         """
+        if not smooth_kws:
+            return self
         # get proper smoothing function based on smooth args
         if isinstance(smooth_kws, str):
             if 'goddard' in smooth_kws:
-                smooth_fct = smooth_goddard_2013
-                smooth_kws = {'lead': 4}  # default
+                if self._datasets['initialized'].lead.attrs['units'] == 'years':
+                    smooth_fct = smooth_goddard_2013
+                    tsmooth_kws = {'lead': 4}  # default
+                    d_lon_lat_kws = {'lon': 5, 'lat': 5}  # default
+                else:
+                    raise ValueError(
+                        '`goddard2013` smoothing only available for annual leads.'
+                    )
             else:
                 raise ValueError(
                     'Please provide from list of available smoothings: \
                      ["goddard2013"]'
                 )
+        # TODO: make independent of lon and lat
```

Comment:
What do you mean by this?

Reply:
The code below actively searches for lot and lat in dims. Maybe this part of the code could be more robust in how it finds these two spatial dimensions regardless of name. Optional work in progress comment.